+itstool (119/548)

Formulae: https://formulae.brew.sh/formula/itstool

Packaging this and running into an issue about the availability of `libxml2`.   My guess is that this python requires the `libxml2` package in the python/lib/site-packages directory, so I tried including a step for `pip install lxml`, but still ran into the same errors.

Here's the error that happens in my `./configure` step:

```sh
+ ./configure --prefix=/Users/sanch/.tea/itstool/v2.0.7 --without-python --without-plugins
configure: WARNING: unrecognized options: --without-python, --without-plugins
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for a Python interpreter with version >= 2.6... python
checking for python... /Users/sanch/.tea/python.org/v3.11.2/bin/python
checking for python version... 3.11
checking for python platform... darwin
checking for python script directory... ${prefix}/lib/python3.11/site-packages
checking for python extension module directory... ${exec_prefix}/lib/python3.11/site-packages
checking for python module libxml2... not found
configure: error: Python module libxml2 is needed to run this package
```

Here's the script and build dependencies in the yaml file:
```yaml
build:
  dependencies:
    tea.xyz/gx/cc: c99
    tea.xyz/gx/make: '*'
    gnome.org/libxml2: '*'
    freedesktop.org/pkg-config: '*'
    # ^^ often packages need dependencies to build, it is very common for
    # packages to need a C compiler and make, the above provide those
  script: |
    ./configure $ARGS
    make --jobs {{ hw.concurrency }}
    make install
  # it’s extremely common for packages to require the above
  env:
    # add any environment variables here
    ARGS:
      - --prefix="{{prefix}}"
      - --without-python
      - --without-plugins 
```

--- a/projects/itstool.org/package.yml
+++ b/projects/itstool.org/package.yml
@@ -13,6 +13,10 @@ dependencies:
   gnome.org/libxml2: '*'
   python.org: =3.11
 
+runtime:
+  env:
+    ITSTOOL_SHARE: "{{prefix}}/share"
+
 build:
   dependencies:
     tea.xyz/gx/cc: c99
@@ -21,6 +25,8 @@ build:
   script:
     - ./configure $ARGS
     - make --jobs {{ hw.concurrency }} install
+    - sed -i 's|DATADIR="[^"]*"|DATADIR="$ITSTOOL_SHARE"|g' {{prefix}}/bin/itstool
+
     
   env:
     ARGS:

--- a/projects/itstool.org/package.yml
+++ b/projects/itstool.org/package.yml
@@ -23,15 +23,13 @@ build:
     tea.xyz/gx/make: '*'
 
   script:
-    - ./configure $ARGS
+    - ./configure $ARGS PYTHON=$(which python3.11)
     - make --jobs {{ hw.concurrency }} install
-    - sed -i 's|DATADIR="[^"]*"|DATADIR="$ITSTOOL_SHARE"|g' {{prefix}}/bin/itstool
+    - sed -i'' 's|DATADIR="[^"]*"|DATADIR="$ITSTOOL_SHARE"|g' {{prefix}}/bin/itstool
 
-    
   env:
     ARGS:
       - --prefix="{{prefix}}"
-      - PYTHON="{{deps.python.org.prefix}}/bin/python3.11"
 
 provides:
   - bin/itstool

--- a/projects/itstool.org/package.yml
+++ b/projects/itstool.org/package.yml
@@ -1,0 +1,33 @@
+distributable:
+  url: http://files.itstool.org/itstool/itstool-{{version}}.tar.bz2
+  strip-components: 1
+
+versions:
+  url: https://itstool.org/download.html
+  match: /itstool-\d+\.\d+\.\d+\.tar\.bz2/
+  strip:
+    - /^itstool-/
+    - /\.tar\.bz2$/
+
+dependencies:
+  gnome.org/libxml2: '*'
+  python.org: =3.11
+
+build:
+  dependencies:
+    tea.xyz/gx/cc: c99
+    tea.xyz/gx/make: '*'
+
+  script:
+    - ./configure $ARGS
+    - make --jobs {{ hw.concurrency }} install
+    
+  env:
+    ARGS:
+      - --prefix="{{prefix}}"
+      - PYTHON="{{deps.python.org.prefix}}/bin/python3.11"
+
+provides:
+  - bin/itstool
+
+test: itstool --version | grep {{version}}

--- a/projects/itstool.org/package.yml
+++ b/projects/itstool.org/package.yml
@@ -13,19 +13,15 @@ dependencies:
   gnome.org/libxml2: '*'
   python.org: =3.11
 
-runtime:
-  env:
-    ITSTOOL_SHARE: "{{prefix}}/share"
-
 build:
   dependencies:
     tea.xyz/gx/cc: c99
     tea.xyz/gx/make: '*'
 
   script:
-    - ./configure $ARGS PYTHON=$(which python3.11)
+    - ./configure $ARGS
     - make --jobs {{ hw.concurrency }} install
-    - sed -i'' 's|DATADIR="[^"]*"|DATADIR="$ITSTOOL_SHARE"|g' {{prefix}}/bin/itstool
+    - sed -i'' -e 's|#!/.* -s|#!/usr/bin/env python|g' {{prefix}}/bin/itstool
 
   env:
     ARGS:

